### PR TITLE
Some feature_delices contributions

### PR DESCRIPTION
**Pull request by chadell**
- Improve local design examples
- Fix the revert of optional dictionary attributes (such as config context) to honor the keys that were added after the design deployment
- Fix `get_design_objects` method
- Add tests for a few functionalities that are not working as expected

### CODE REVIEW

File: examples/custom_design/designs/core_site/jobs.py [hunk 1]
```diff
@@ -28,3 +28,6 @@ class Meta:
         commit_default = False
         design_file = "designs/0001_design.yaml.j2"
         context_class = CoreSiteContext
+        version = "0.2.5"
```
Comment (abates) [inline examples/custom_design/designs/core_site/jobs.py:31]: This is new... are we leveraging the design version yet?

Comment (chadell) [reply examples/custom_design/designs/core_site/jobs.py:31]: only informational for now, but not used to update the design yet.

File: nautobot_design_builder/migrations/0001_initial.py
Comment (abates) [inline nautobot_design_builder/migrations/0001_initial.py:1]: Are these changes just auto-formatting changes, or new/squashed migration?

File: nautobot_design_builder/tests/test_design_job.py [hunk 2]
```diff
@@ -194,8 +191,35 @@ def setUp(self):
     def test_create_integration_design(self):
```
Comment (abates) [inline nautobot_design_builder/tests/test_design_job.py:191]: Is this now the failing test that demonstrates I need to fix the cable and next prefix extensions?

Comment (chadell) [reply nautobot_design_builder/tests/test_design_job.py:191]: no, the ones that fail have been skipped with `@unittest.skip` until are fixed (below)

File: nautobot_design_builder/migrations/0001_initial.py
Comment (chadell) [reply nautobot_design_builder/migrations/0001_initial.py:1]: just formating

**Pull request merged**